dev/tool/src/elastic.ts

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright © 2020, 2021 Anticrm Platform Contributors.
+// Copyright © 2021 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import core, { Account, Doc, DOMAIN_TX, generateId, Hierarchy, ModelDb, Ref, Tx } from '@anticrm/core'
+import { Client as ElasticClient } from '@elastic/elasticsearch'
+import { Db, MongoClient } from 'mongodb'
+import { Client } from 'minio'
+import { listMinioObjects } from './minio'
+import { createElasticAdapter } from '@anticrm/elastic'
+import { FullTextAdapter, FullTextIndex, IndexedDoc } from '@anticrm/server-core'
+import { DOMAIN_ATTACHMENT } from '@anticrm/model-attachment'
+import { createMongoAdapter } from '@anticrm/mongo'
+
+export async function rebuildElastic (
+  mongoUrl: string,
+  dbName: string,
+  minio: Client,
+  elasticUrl: string
+): Promise<void> {
+  await dropElastic(elasticUrl, dbName)
+  return await restoreElastic(mongoUrl, dbName, minio, elasticUrl)
+}
+
+async function dropElastic (elasticUrl: string, dbName: string): Promise<void> {
+  return await new Promise((resolve, reject) => {
+    const client = new ElasticClient({
+      node: elasticUrl
+    })
+    client.indices.exists(
+      {
+        index: dbName
+      },
+      (err: any, result: any) => {
+        if (err != null) reject(err)
+        if (result.body === true) {
+          client.indices.delete(
+            {
+              index: dbName
+            },
+            (err: any, result: any) => {
+              if (err != null) reject(err)
+              resolve(result)
+            }
+          )
+        } else {
+          resolve()
+        }
+      }
+    )
+  })
+}
+
+async function restoreElastic (mongoUrl: string, dbName: string, minio: Client, elasticUrl: string): Promise<void> {
+  const mongoClient = new MongoClient(mongoUrl)
+  try {
+    await mongoClient.connect()
+    const db = mongoClient.db(dbName)
+    const elastic = await createElasticAdapter(elasticUrl, dbName)
+    const fullTextIndex = await createFullTextIndex(db, elastic, mongoUrl, dbName)
+    const txes = (await db.collection(DOMAIN_TX).find().toArray()) as Tx[]
+    for (const tx of txes) {
+      try {
+        await fullTextIndex.tx(tx)
```

Comment:
Should we process derived data as well here as well? Please see how it is done in TServerStorage: https://github.com/hcengineering/anticrm/blob/29a63e838cb3f6af0d78bd05cea4c20bf0d73eaa/server/core/src/storage.ts#L159